mdn/developer-portal
# 624: Rename Articles to Posts

This changeset renames Articles to Posts -- but only at points where humans see them. 

This is a lot smaller changeset than would be required to rename the `Article` and `Articles` models, which may still be needed, but we're currently just doing the smallest changeset possible -- partly because the name may change again.
 
(Resolves #624)

## Key changes:

- Soft-relabel Article as Post and Articles as Posts (in terms of the Wagtail UI)
- Add a data migration that renames "Articles" to "Posts" and updates all relevant slugs from /articles/* to /posts/*
- Relabel External Articles as External posts
- Surface-level rephrases from article* to post* (HTML templates, RSS feed, one FormField help-text)
- Changes to non-Article models models, renaming referencs to 'article*' to 'post*' where relevant, for consistent Wagtail UI. **NOTE THAT THIS WILL BREAK STREAMFIELD CONTENT IN STAGING**
- Update the featured.html template to check for 'post' as a block type, not 'article' in order to ensure that featured posts are rendered
- Update common.json fixture (used in tests and staging setup) to reflect the relabelling of article* to post*
- Add a small document explaining the soft-relabelling of Article* to Post*
- Fix test regressions caused by other items in this changeset

*Important*: the data in StreamFields is essentially JSON, and the model change here to the StreamField spec means that exisiting JSON in Staging that references `article` as the  StreamBlock type will get ignored because we've changed the configuration to label the block as a `post` now, for UI constency. I could have added a data migration, but for Staging data it did not seem worthwhile. Please say if you disagree.

## How to test

- check out `master`
- `docker-compose build && docker-compose up`
- view the site and see that:
  - the menu says Articles
  - paths to articles are /articles/
  - the wagtail UI only talks in terms of Articles
- check out this branch
- `docker-compose exec app python manage.py migrate` to run migrations
- view the site and see that:
  - the menu says Posts
  - paths to articles are /posts/
  - the wagtail UI only talks in terms of Posts
  - you can edit the Homepage (for example) and add a Post as 'Featured' and it'll appear when you preview/publish the homepage

## Screenshots


Title and slug change for what was once `/articles/`

![Screenshot 2019-11-01 at 15 14 02](https://user-images.githubusercontent.com/101457/68034844-5ca5d800-fcba-11e9-9426-97fa9f62d482.png)
![Screenshot 2019-11-01 at 15 14 06](https://user-images.githubusercontent.com/101457/68034845-5ca5d800-fcba-11e9-8bcc-a0c089ae37e6.png)

----

External Articles now called External Posts

![Screenshot 2019-11-01 at 11 51 14](https://user-images.githubusercontent.com/101457/68034858-67f90380-fcba-11e9-8185-fbd261d6bc68.png)
![Screenshot 2019-11-01 at 11 51 19](https://user-images.githubusercontent.com/101457/68034859-67f90380-fcba-11e9-91da-bc6b967b40f1.png)

----
Topics now talk in terms of posts

![Screenshot 2019-11-01 at 13 44 55](https://user-images.githubusercontent.com/101457/68034892-79daa680-fcba-11e9-9174-c3552f651cd9.png)

As does the Person page
![Screenshot 2019-11-01 at 15 18 13](https://user-images.githubusercontent.com/101457/68035140-f077a400-fcba-11e9-90fb-61f1739db9ae.png)



Posts are definitely under `/posts/`

![Screenshot 2019-11-01 at 13 44 37](https://user-images.githubusercontent.com/101457/68034891-79daa680-fcba-11e9-81d3-4076c17d3b2b.png)
